UI: Deduplication of photo cards, mosaic and list view components

heikomat opened the pull request:
This dedupliactes the mosiac- cards- and list views.

These views existed twice, once for the regular search and once for shared albums.
Both instances of these views were nearly identical. The views for the shared albums had some features removed, and in some pleaces used `div`s instead of `button`s.

This should somewhat improve maintainability, because fixind/changing something in one of them no longer required the developer to remember to also apply these changes to the corresponding other view. having duplicate code for something that is supposed to behave identical increases the risk of errors.

This is "fully implemented" but i'll need to have a last look around to see if anything else needs to be adjusted (for example translations) and run the tests. Will do that when i'm less tired 😅

Acceptance Criteria:

- [x] **Features and improvements are fully implemented** so that they can be released at any time without additional work
- [ ] **Automated unit and/or acceptance tests have been added** to ensure the changes work as expected and to reduce repetitive manual work
- [ ] **User interface changes are fully responsive** and have been tested on all major browsers and various devices
- [x] Database-related changes are compatible with SQLite and MariaDB
- [ ] Translations have been / will be updated (specify if needed)
- [ ] Documentation has been / will be updated (specify if needed)
- [x] Contributor License Agreement (CLA) has been signed

Commit:
virtualize mosaic-view

Commit:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

Commit:
start virtualizing parts of the list view

Commit:
drastically improve mosaic component updating performance by not rendering cards if not necessary

Commit:
speed up list view virtualization by preventing the rendering of vue-components

Commit:
start virtualizing cards view

Commit:
continue virtualizing card view

Commit:
finish virtualizing cards

Commit:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

Commit:
start moving common virtualization logic into shared file

Commit:
try speeding up access to calculated photo details

Commit:
remove console.log

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

Commit:
improve accuracy of cards-placeholder elements

Commit:
remove console.log

Commit:
start fixing memoized values not updating on change

heikomat committed:
fixing memoized values not updating on change

heikomat committed:
remove console.logs

heikomat committed:
fix getting location info after memoizing the function

heikomat committed:
remove obsolete comment

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
start rendering of only visible icons in mosaic-view

heikomat committed:
continue rendering of only visible icons in mosaic-view

heikomat committed:
implement rendering of only visible icons in cards-view

heikomat committed:
implement rendering of only visible icons in list-view

heikomat committed:
memoize some photo-model functions that are called when rendering mosaic- or cards-view

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
fix cards sometimes getting smaller when scrolling

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
improve fix for cards sometimes getting smaller when scrolling

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
deduplicate mosaic-, cards- and list-view

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/all-virtualization

heikomat committed:
prevent rendering of no-photos-alert if photos are present

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization_dedupe-views' into feature/all-virtualization

heikomat committed:
fix selection issues after implementing conditional button-rendering

heikomat committed:
speedup mosaic rendering by replacing vue components with html components

heikomat committed:
speedup cards rendering by replacing vue components with html components

heikomat committed:
speedup list rendering by replacing vue components with html components

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/all-virtualization

heikomat committed:
fix removed elements leaving behind placeholders in view

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/all-virtualization

heikomat committed:
speedup photo view rendering by replacing vue buttons and icons with regular html components

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/all-virtualization

heikomat committed:
fix positioning of card-title in placeholder elements

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/all-virtualization

heikomat committed:
fix missing icons after multiselect or select via touch

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/all-virtualization

heikomat committed:
prevent flickering of favourite button on click

heikomat committed:
prevent flickering of favourite button on click

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/all-virtualization

heikomat committed:
use div instead of v-flex

heikomat committed:
replace inline styles with css classes

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/all-virtualization

heikomat committed:
re-add actually necessary css-classes

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/all-virtualization

heikomat committed:
add size-containment to mosaic-elements to reduce re-layouting costs when virtualization replaces children

heikomat committed:
fix typo in comment

heikomat committed:
use plain html instead of components and css-classes instead of inline styles in cards.vue

heikomat committed:
improve list scrolling by not using v-data-table for a static table

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/all-virtualization

heikomat committed:
fix icon-color on card-details in light-theme

heikomat committed:
fix card-details showing wrong icons with wrong height on pixel 3

heikomat committed:
fix animated gif-previews getting out of image-bounds when hovering in firefox

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
fix closing brackets in mosaic view not matching opening brackets

heikomat committed:
fix live-photo icon in mosaic- cards and list components

heikomat committed:
improve render performance by replacing v-hover vue-component with css-hidden button

heikomat committed:
prevent unnecessary rerenders

heikomat committed:
prevent unnecessary rerenders

heikomat committed:
undo "prevent unnecessary rerender" because the real-world-effect is negligable

heikomat committed:
load next batch earlier to reduce change of scrolling to the end before the load finished

heikomat committed:
add explanation on why the selection-button isnt removed via v-if

heikomat committed:
remove console.log

heikomat committed:
speed up rendering by reducing amount of observed items

heikomat committed:
fix favourite-buttons in non-search views

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
prevent unnecessary observeItems-calls by only observing items when photos changed

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
fix typo in isSharedView

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/all-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization_dedupe-views' into feature/all-virtualization

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
fix like-button being shown on shared albums

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/photo-view-virtualization

heikomat committed:
Merge branch 'feature/photo-view-virtualization' into feature/mosaic-element-render-performance

heikomat committed:
Merge remote-tracking branch 'upstream/develop' into feature/mosaic-element-render-performance

heikomat committed:
Merge branch 'feature/mosaic-element-render-performance' into feature/photo-view-virtualization_dedupe-views

lastzero commented:
Is there anything specific we should/need to do so we can merge this next week? Theresa is out of the office until Monday. I don't know if she has started testing yet...

heikomat commented:
> Is there anything specific we should/need to do so we can merge this next week? Theresa is out of the office until Monday. I don't know if she has started testing yet...

I need to merge upstram develop again, as I haven't done this in a while.
After that I'll do a little more testing myself. Then it should be ready to go. I might do that tomorrow morning

heikomat committed:
Merge branch 'develop' into feature/photo-view-virtualization_dedupe-views

heikomat committed:
increase preloaded window-height on regular album-view and shared album-view to match search

heikomat commented:
@lastzero i think this is ready for preview now, except for one thing:
i deleted components and added texts to existing components. It seems to work fine, but i have no idea whatsoever if some translations need to be cleaned up or added or something like this, as i have no experience with how photoprism handles translations.

lastzero commented:
Did you check if the changes have any unexpected performance impact? Just to be sure...

heikomat commented:
> Did you check if the changes have any unexpected performance impact? Just to be sure... 

I did one quick performance profiling, and nothing looked out of the ordinary.
Wouldn't have expected a noticable impact though, because it just adds one boolean and skips rendering some things and adding some listeners if it is set

heikomat committed:
prevent showing merged results in shared view

heikomat committed:
Revert "prevent showing merged results in shared view"

This reverts commit c5321af8bc25be9a3f0710720ec4773ef6f688dc.

heikomat commented:
wait a second before merging, i might have found a discrepenccy in the code that i didn't notice in testing, regarding the third paramter of `onOpen`

heikomat commented:
@lastzero quick question about the third parameter of onOpen. its usage seems to be inconsistent.

[Looking at the search-component](https://github.com/photoprism/photoprism/blob/519f0c49c9ec139632a7b0031cadfcf91564f448/frontend/src/component/photo/cards.vue#L94-L115), it is always true, except when clicking or touching the zoom-button.

[Looking at the shared-album-component](https://github.com/photoprism/photoprism/blob/519f0c49c9ec139632a7b0031cadfcf91564f448/frontend/src/share/photo/cards.vue#L52-L81) it is always true, except when clicking the zoom-icon, but not when touching the zoom icon. Is this intended?

My feeling tells me, that when clicking the zoom icon in the shared-album-view, that parameter should also be `false`. Is that assumption correct?

lastzero commented:
Well done! Changes seem straightforward for the most part, with the exception of this event handler, which looks pretty fancy... how did you come up with this construct? Just out of curiosity. If it works, why not!? 🧐

```
@[!isSharedView&&`click`].exact="editPhoto(index)"
```

heikomat committed:
make click on top-left icon open photo with showMerged=true

heikomat commented:
> !isSharedView&&`click`

I found it here: https://stackoverflow.com/a/57235465/3213368

interestingly enough, it seems to only work when written exactly like that. These variants don't work, and i don't know why:

```javascript
// notice the nspacing around the &&
@[!isSharedView && `click`].exact="editPhoto(index)"

// notice the '.exact' being withing the brackets
@[!isSharedView&&`click.exact`]="editPhoto(index)"
```

lastzero commented:
> @lastzero quick question about the third parameter of onOpen. its usage seems to be inconsistent.

It determines whether all images of only a single stack are displayed, or each primary image of the search results. By default, the fullscreen viewer does not also cycle through all images in stacks. Only the primary images are displayed, depending on which one is active.

Otherwise there would be no way to view the other images in a stack, except perhaps as small thumbnails in the Edit dialog. Or you would be forced to always cycle through the often very similar images in stacks and you could not select a best picture.

This is why certain buttons, especially icons like the RAW and photo stack icons, open the viewer differently. We also assumed that browsing through the images in stacks is usually unwanted when sharing, since the instance owner should have already selected the highest quality image when sharing it.

heikomat commented:
> > @lastzero quick question about the third parameter of onOpen. its usage seems to be inconsistent.
> 
> It determines whether all images of only a single stack are displayed, or each primary image of the search results. By default, the fullscreen viewer does not also cycle through all images in stacks. Only the primary images are displayed, depending on which one is active.
> 
> Otherwise there would be no way to view the other images in a stack, except perhaps as small thumbnails in the Edit dialog. Or you would be forced to always cycle through the often very similar images in stacks and you could not select a best picture.
> 
> This is why certain buttons, especially icons like the RAW and photo stack icons, open the viewer differently. We also assumed that browsing through the images in stacks is usually unwanted when sharing, since the instance owner should have already selected the highest quality image when sharing it.

ok, so this third parameter should basically always be false, except when clicking on certain icons in the non-shared view. Or in other words, stacks should never be opened in the shared view. is that correct?

If so, the current implementation of shared views is not correct, but i can fix that in this PR

lastzero commented:
> in other words, stacks should never be opened in the shared view. is that correct?

This seems to be correct, unless I forgot an edge case. The functionality is generally not needed for sharing. **Edit: It might also be "true" in share mode if it is not an image but a video, which is then opened in the video player. It's possible that the handler function automatically catches this, but I'm not 100% sure. Best to check the actual code and test it.**

Also note that clicking on a picture does not open it in selection mode, i.e. when at least one item is selected. The magnifying glass icon in the upper left corner is then displayed so that pictures can still be opened in full screen mode, as the preview images may not show enough detail to make a choice.

lastzero commented:
This is to clarify what I meant with *It might also be "true" in share mode if it is not an image but a video, which is then opened in the video player.* 👇

![Screenshot 2022-06-24 at 20 51 35](https://user-images.githubusercontent.com/301686/175647349-acc8ab95-fe37-48ec-83c1-f09fa3782bed.png)

**It determines, for example, whether a Live Photo - which contains an image and a video - is opened in the photo viewer or in the video player.**

heikomat commented:
ok, then that is also currently not correctly implemented everywhere.

Give me a couple of minutes to straighten that up :)

lastzero commented:
Take your time and check my updated comment at the bottom, as it is especially important for Live Photos 👍

heikomat commented:
@lastzero somewhat counterintuitively, this condition:
```javascript
if (showMerged && selected.Type === MediaLive || selected.Type === MediaVideo || selected.Type === MediaAnimated) {
```
Makes it so that `MediaVideo` and `MediaAnimated` are always opened in a video player, no matter what. Only `MediaLive` is opened in a videoplayer, if `showMerged` is true, because there are no braces around the Media-Type-Checks.

i guess from an enduser perspective, that is a valid behaviour. i just want to ask: Is that by design or by accident? Is this specific condition working as intended, or is it incorrect because of missing braces?

lastzero commented:
> i guess from an enduser perspective, that is a valid behaviour. i just want to ask: Is that by design or by accident? Is this specific condition working as intended, or is it incorrect because of missing braces?

- This is intentional, because Live Photos play in the search results on mouse over. When you click on the picture (and not on the Live Icon in the upper left corner), the high-resolution image opens and not the video, which may have a lower resolution.
- Regular videos, on the other hand, always open in the video player. Therefore, the flag is not checked in this case. Also, the preview image is extracted from the video and thus always has the same resolution (edit: same for animated GIFs).
- Only when you click the Live icon on Live Photos in the search results, the video will open in the player.

*Live Photos is a feature we implemented specifically for our users, since we don't have any ourselves. It turned out to be the most intuitive solution for them. As far as I know, users are happy with it, so we should keep it.*

lastzero commented:
Feel free to add comments with an excerpt of this information so that the code is less of a mystery next time!

heikomat commented:
> Feel free to add comments with an excerpt of this information so that the code is less of a mystery next time!

already on it ;)

heikomat committed:
make decision as to what player to open on openPhoto more consistent

heikomat commented:
@lastzero here, [that](https://github.com/photoprism/photoprism/pull/2434/commits/9120bdca9e7992496ad27087eb6282f9d2629cf9) should do it. What do you think?

lastzero commented:
I see that the function signature has changed....
- Even if it is not passed to components that do not expect it (I assume you checked), there could possibly be similar functions in other components, so the signature/implementation might now differ across the application and cannot be easily compared.
- If there is a compelling reason to do it this way, then other places should be refactored as well, in general.
- At the moment, I'm not sure what's best.... I suppose you have good reasons to change it this way... see chat :)

heikomat commented:
> I see that the function signature has changed....
> 
>     * Even if it is not passed to components that do not expect it (I assume you checked), there could possibly be similar functions in other components, so the signature/implementation might now differ across the application and cannot be easily compared.
> 
>     * If there is a compelling reason to do it this way, then other places should be refactored as well, in general.
> 
>     * At the moment, I'm not sure what's best.... I suppose you have good reasons to change it this way... see chat :)

As you said, if there are similiar, comparable functions we could consider adjusting their parameters aswell. Although i wouldn't know what functions that might be, and if there even are functions like this, or if this is a fully hypothetical case.
   I did adjust all the `onOpen` and `openPhoto` calls, and a `showMerged`-parameter only exists in those. Before this change, the effect of that paramter was not clear, because it served multiple purposes in different scenarios.
   That also led to it being used wrong in some places. (at some places regarding videos it was explicitly set to true, even though it didn't make a difference, because videos are always opened in the video-player anyway)
   
 In other words: I'm somewhat confident, that that change doesn't make the code worse in any aspect (including comparibility).
 I therefore think this is mergable, because even if there might be ways to further increase code-quality by (for example) using more, smaller functions (as you mentioned in chat), as long as it doesn't make anything worse, it's a win.

heikomat commented:
@lastzero can this get merged? If not, what is missing?

lastzero commented:
Just time :D I merged a video transcoding PR on Friday which kept me busy until yesterday evening (see commits). Now I have other work to do that I had planned for the weekend. I'll get back to you as soon as possible :+1:

lastzero commented:
**Thanks a lot for your patience and all the work you put into this!** 💎

I'm going to merge it now, since there's no point in changing it back and forth and we need to test it completely anyway.

In general, I would have preferred not to change this function, simply to reduce diff complexity, save review/testing time, and avoid unexpected glitches.

Slight inconsistencies have already slipped into this PR when you compare the function signature between the photo search and the album view:

1. `openPhoto(index, showMerged, preferVideo = false)`
2. `openPhoto(index, showMerged = false, preferVideo = false)`

heikomat commented:
> **Thanks a lot for your patience and all the work you put into this!** 💎
> 
> I'm going to merge it now, since there's no point in changing it back and forth and we need to test it completely anyway.
> 
> In general, I would have preferred not to change this function, simply to reduce diff complexity, save review/testing time, and avoid unexpected glitches.
> 
> Slight inconsistencies have already slipped into this PR when you compare the function signature between the photo search and the album view:
> 
> 1. `openPhoto(index, showMerged, preferVideo = false)`
> 2. `openPhoto(index, showMerged = false, preferVideo = false)`

Oh, good catch. That shouldn't make a difference at runtime, but having them differ was indeed not on purpose

lastzero commented:
There is another tiny "glitch" that I can't explain..... At first I thought a dependency changed, but I see you also added some "memoizeOne" magic to the `Photo.getDateString(showTimeZone)` function (not necessarily in this PR, but it's a recent change):

- Original Date String: `Sunday, July 8, 2012, 2:45 PM`
- New Date String:    `Sunday, July 8, 2012 at 2:45 PM`
- Related Test: [frontend/tests/unit/model/photo_test.js](https://github.com/photoprism/photoprism/commit/161fa2b327d83f2cb2d91b41910968e5fdb684ee#diff-b461acf7879636ab98a0e7ecfcfa058986094e49b4f976cb5cfdac41dd06dae9)

The test failed the first time for me today, but that doesn't mean that the return value didn't change before and I just didn't notice it.

Any idea?

Initial exploratory UI tests haven't revealed anything unexpected so far, which is great. @graciousgrey will do more testing tomorrow. Automated acceptance tests might already be running, unless the computer crashed due to overheating.

lastzero commented:
Our [development preview](https://docs.photoprism.app/release-notes/#development-preview) and [demo.photoprism.app](https://demo.photoprism.app/) have been updated so everyone can help test these changes! :birthday:

heikomat commented:
> There is another tiny "glitch" that I can't explain..... At first I thought a dependency changed, but I see you also added some "memoizeOne" magic to the `Photo.getDateString(showTimeZone)` function (not necessarily in this PR, but it's a recent change):
> 
>     * Original Date String: `Sunday, July 8, 2012, 2:45 PM`
> 
>     * New Date String:    `Sunday, July 8, 2012 at 2:45 PM`
> 
>     * Related Test: [frontend/tests/unit/model/photo_test.js](https://github.com/photoprism/photoprism/commit/161fa2b327d83f2cb2d91b41910968e5fdb684ee#diff-b461acf7879636ab98a0e7ecfcfa058986094e49b4f976cb5cfdac41dd06dae9)
> 
> 
> The test failed the first time for me today, but that doesn't mean that the return value didn't change before and I just didn't notice it.
> 
> Any idea?
> 
> Initial exploratory UI tests haven't revealed anything unexpected so far, which is great. @graciousgrey will do more testing tomorrow. Automated acceptance tests might already be running, unless the computer crashed due to overheating.

I'll look into it. It does ring a bell. i very vaguely remember something like this

lastzero commented:
My instinct was correct, the acceptance tests crashed due to overheating! 🤒

heikomat commented:
@lastzero the commit you linked made the tests fail on my laptop:

```
asset commons.js 3.25 MiB [emitted] (name: commons) (id hint: commons)
asset runtime.js 7.62 KiB [emitted] (name: runtime)
asset config-options_test.532879671.js 1.03 KiB [emitted] (name: config-options_test.532879671)
asset clipboard_test.2297578351.js 1.03 KiB [emitted] (name: clipboard_test.2297578351)
asset settings_test.1305824515.js 1.02 KiB [emitted] (name: settings_test.1305824515)
asset caniuse_test.4219863473.js 1.02 KiB [emitted] (name: caniuse_test.4219863473)
asset options_test.994661877.js 1.02 KiB [emitted] (name: options_test.994661877)
asset session_test.3602554901.js 1.02 KiB [emitted] (name: session_test.3602554901)
asset subject_test.1348792091.js 1.02 KiB [emitted] (name: subject_test.1348792091)
asset account_test.976279458.js 1.02 KiB [emitted] (name: account_test.976279458)
asset viewer_test.2738163870.js 1.02 KiB [emitted] (name: viewer_test.2738163870)
asset config_test.304515869.js 1.02 KiB [emitted] (name: config_test.304515869)
asset folder_test.1045800480.js 1.02 KiB [emitted] (name: folder_test.1045800480)
asset notify_test.649733240.js 1.02 KiB [emitted] (name: notify_test.649733240)
+ 13 assets
webpack 5.73.0 compiled successfully in 8211 ms
Chrome Headless 100.0.4896.88 (Linux x86_64) model/photo should get date string FAILED
	AssertionError: expected 'Sunday, July 8, 2012, 2:45 PM' to equal 'Sunday, July 8, 2012 at 2:45 PM'
	    at Context.eval (webpack://photoprism/./tests/unit/model/photo_test.js?:289:12)
Chrome Headless 100.0.4896.88 (Linux x86_64): Executed 319 of 319 (1 FAILED) (0.543 secs / 0.238 secs)
TOTAL: 1 FAILED, 318 SUCCESS
TOTAL: 1 FAILED, 318 SUCCESS
```

So on my machine, **it does still produce the old string** (without the `at`).
I've got two possible explanations:

- time-config on either your or my machine
- dependency-updates (i rebuilt the frontend, but did not re-install the dependencies after pulling `origin/develop`

Will check with reinstalled dependencies, but that might take a while, as `npm install` takes forever when done from within the docker-container on a macbook

heikomat commented:
@lastzero update: even after fully reinstalling all dependencies, i still get the old string on my machine, meaning that the tests now fail for me after your change to the test

lastzero commented:
Are you running the tests in our latest Docker "develop" image? If you run it on your local OS, test results that depend on your locale or timezone settings may fail. Note that you need to explicitly build a new local image using `docker-compose build`. For JS deps, `npm ci` seems to be the cleanest solution to get exactly the same packages. Our Makefile should already do that if you simply run `make all` in the container after building the develop image.

heikomat commented:
> Are you running the tests in our latest Docker "develop" image? If you run it on your local OS, test results that depend on your locale or timezone settings may fail. Note that you need to explicitly build a new local image using `docker-compose build`. For JS deps, `npm ci` seems to be the cleanest solution to get exactly the same packages. Our Makefile should already do that if you simply run `make all` in the container after building the develop image.

I did not rebuild the image, but I did run `make clean`, `make all`, `cd frontend`, `npm run build` and `npm run test` all within the docket container

lastzero commented:
Yes, so if the result is changed by a Debian update you don't have, that would explain the differences.

heikomat commented:
> Yes, so if the result is changed by a Debian update you don't have, that would explain the differences.

Will check that tomorrow.
In that case it would not be a change caused by faulty memorization though :D

heikomat commented:
@lastzero the culprit is definitly a package that is installed via apt that has been updated since `220415-bookworm`.

Instead of outright updating the whole image, i tried `sudo apt update` and `sudo apt upgrade`, which indeed made `getDateString` return the new format (the one with `at`)

Before updating though, i listed the outdated packages. So one of these must be at fault here
<details>
  <summary>oudated packages (click me)</summary>
  
```
adwaita-icon-theme/testing 42.0-2 all [upgradable from: 42~really41.0-1]
apache2-utils/testing 2.4.54-1 amd64 [upgradable from: 2.4.53-2]
apt-utils/testing 2.5.0 amd64 [upgradable from: 2.4.5]
apt/testing 2.5.0 amd64 [upgradable from: 2.4.5]
bash/testing 5.1-6.1 amd64 [upgradable from: 5.1-6]
binutils-common/testing 2.38-4 amd64 [upgradable from: 2.38-3]
binutils-x86-64-linux-gnu/testing 2.38-4 amd64 [upgradable from: 2.38-3]
binutils/testing 2.38-4 amd64 [upgradable from: 2.38-3]
bsdutils/testing 1:2.38-4 amd64 [upgradable from: 1:2.37.3-1+b1]
chromium-common/testing 103.0.5060.53-1 amd64 [upgradable from: 100.0.4896.88-1]
chromium-driver/testing 103.0.5060.53-1 amd64 [upgradable from: 100.0.4896.88-1]
chromium-sandbox/testing 103.0.5060.53-1 amd64 [upgradable from: 100.0.4896.88-1]
chromium/testing 103.0.5060.53-1 amd64 [upgradable from: 100.0.4896.88-1]
cpp-11/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
curl/testing 7.83.1-2 amd64 [upgradable from: 7.82.0-2]
darktable/testing 3.8.1-2+b1 amd64 [upgradable from: 3.8.1-2]
debianutils/testing 5.7-0.2 amd64 [upgradable from: 5.7-0.1]
distro-info-data/testing 0.53 all [upgradable from: 0.52]
dpkg-dev/testing 1.21.8 all [upgradable from: 1.21.7]
dpkg/testing 1.21.8 amd64 [upgradable from: 1.21.7]
ffmpeg/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
findutils/testing 4.9.0-3 amd64 [upgradable from: 4.9.0-2]
firefox-esr/testing 91.10.0esr-1 amd64 [upgradable from: 91.8.0esr-1]
g++-11/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
gcc-11-base/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
gcc-11/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
gcc-12-base/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
gir1.2-glib-2.0/testing 1.72.0-1+b1 amd64 [upgradable from: 1.72.0-1]
gpg/testing 2.2.35-2 amd64 [upgradable from: 2.2.27-3+b1]
gpgconf/testing 2.2.35-2 amd64 [upgradable from: 2.2.27-3+b1]
gpgv/testing 2.2.35-2 amd64 [upgradable from: 2.2.27-3+b1]
gtk-update-icon-cache/testing 3.24.34-1 amd64 [upgradable from: 3.24.33-1]
hdf5-helpers/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
icu-devtools/testing 71.1-3 amd64 [upgradable from: 67.1-7]
init-system-helpers/testing 1.63 all [upgradable from: 1.62]
iso-codes/testing 4.10.0-1 all [upgradable from: 4.9.0-1]
krb5-multidev/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libabsl20210324/testing 0~20210324.2-4 amd64 [upgradable from: 0~20210324.2-2]
libaom3/testing 3.4.0-1 amd64 [upgradable from: 3.3.0-1]
libaprutil1/testing 1.6.1-5+b2 amd64 [upgradable from: 1.6.1-5+b1]
libapt-pkg6.0/testing 2.5.0 amd64 [upgradable from: 2.4.5]
libasan6/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
libass9/testing 1:0.16.0-1 amd64 [upgradable from: 1:0.15.2-1]
libassuan0/testing 2.5.5-3 amd64 [upgradable from: 2.5.5-1]
libatomic1/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libatspi2.0-0/testing 2.44.1-1 amd64 [upgradable from: 2.44.0-3]
libavahi-client3/testing 0.8-6 amd64 [upgradable from: 0.8-5]
libavahi-common-data/testing 0.8-6 amd64 [upgradable from: 0.8-5]
libavahi-common3/testing 0.8-6 amd64 [upgradable from: 0.8-5]
libavcodec-extra58/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libavcodec-extra/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libavdevice58/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libavfilter7/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libavformat58/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libavutil56/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libbinutils/testing 2.38-4 amd64 [upgradable from: 2.38-3]
libblas3/testing 3.10.1-2 amd64 [upgradable from: 3.10.0-2]
libblkid1/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
libbluray2/testing 1:1.3.1-2 amd64 [upgradable from: 1:1.3.1-1]
libcaca0/testing 0.99.beta20-2 amd64 [upgradable from: 0.99.beta19-2.2]
libcap-ng0/testing 0.8.3-1 amd64 [upgradable from: 0.7.9-2.2+b1]
libcc1-0/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libcryptsetup12/testing 2:2.4.3-1+b1 amd64 [upgradable from: 2:2.4.3-1]
libctf-nobfd0/testing 2.38-4 amd64 [upgradable from: 2.38-3]
libctf0/testing 2.38-4 amd64 [upgradable from: 2.38-3]
libcups2/testing 2.4.2-1 amd64 [upgradable from: 2.4.1op1-2]
libcurl3-gnutls/testing 7.83.1-2 amd64 [upgradable from: 7.82.0-2]
libcurl4-openssl-dev/testing 7.83.1-2 amd64 [upgradable from: 7.82.0-2]
libcurl4/testing 7.83.1-2 amd64 [upgradable from: 7.82.0-2]
libdb5.3/testing 5.3.28+dfsg1-0.9 amd64 [upgradable from: 5.3.28+dfsg1-0.8]
libdebconfclient0/testing 0.263 amd64 [upgradable from: 0.261]
libdeflate0/testing 1.12-1 amd64 [upgradable from: 1.10-2]
libdpkg-perl/testing 1.21.8 all [upgradable from: 1.21.7]
libdvdread8/testing 6.1.3-1 amd64 [upgradable from: 6.1.2-1]
libelf1/testing 0.187-1 amd64 [upgradable from: 0.186-1]
libevent-2.1-7/testing 2.1.12-stable-5+b1 amd64 [upgradable from: 2.1.12-stable-3]
libfdisk1/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
libflac8/testing 1.3.4-2 amd64 [upgradable from: 1.3.4-1]
libfreetype-dev/testing 2.12.1+dfsg-3 amd64 [upgradable from: 2.11.1+dfsg-1]
libfreetype6-dev/testing 2.12.1+dfsg-3 amd64 [upgradable from: 2.11.1+dfsg-1]
libfreetype6/testing 2.12.1+dfsg-3 amd64 [upgradable from: 2.11.1+dfsg-1]
libgbm1/testing 22.0.5-1 amd64 [upgradable from: 21.3.8-1]
libgcc-11-dev/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
libgcc-s1/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libgfortran5/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libgirepository-1.0-1/testing 1.72.0-1+b1 amd64 [upgradable from: 1.72.0-1]
libgl1-mesa-dri/testing 22.0.5-1 amd64 [upgradable from: 21.3.8-1]
libglapi-mesa/testing 22.0.5-1 amd64 [upgradable from: 21.3.8-1]
libglib2.0-0/testing 2.72.2-2 amd64 [upgradable from: 2.72.0-1+b1]
libglibmm-2.4-1v5/testing 2.66.4-1 amd64 [upgradable from: 2.66.2-2+b1]
libglx-mesa0/testing 22.0.5-1 amd64 [upgradable from: 21.3.8-1]
libgmp10/testing 2:6.2.1+dfsg1-1 amd64 [upgradable from: 2:6.2.1+dfsg-3]
libgnutls30/testing 3.7.6-2 amd64 [upgradable from: 3.7.3-4+b1]
libgomp1/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libgpg-error0/testing 1.45-2 amd64 [upgradable from: 1.43-3]
libgphoto2-6/testing 2.5.29-1 amd64 [upgradable from: 2.5.27-1]
libgphoto2-port12/testing 2.5.29-1 amd64 [upgradable from: 2.5.27-1]
libgsm1/testing 1.0.19-2 amd64 [upgradable from: 1.0.19-1]
libgssapi-krb5-2/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libgssrpc4/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libgtk-3-0/testing 3.24.34-1 amd64 [upgradable from: 3.24.33-1]
libgtk-3-common/testing 3.24.34-1 all [upgradable from: 3.24.33-1]
libgtkmm-3.0-1v5/testing 3.24.6-1 amd64 [upgradable from: 3.24.5-1]
libharfbuzz0b/testing 2.7.4-1+b1 amd64 [upgradable from: 2.7.4-1]
libhdf5-103-1/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libhdf5-cpp-103-1/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libhdf5-dev/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libhdf5-fortran-102/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libhdf5-hl-100/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libhdf5-hl-cpp-100/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libhdf5-hl-fortran-100/testing 1.10.7+repack-4+b1 amd64 [upgradable from: 1.10.7+repack-4]
libheif-examples/testing 1.12.0-2+b4 amd64 [upgradable from: 1.12.0-2+b3]
libheif1/testing 1.12.0-2+b4 amd64 [upgradable from: 1.12.0-2+b3]
libicu-dev/testing 71.1-3 amd64 [upgradable from: 67.1-7]
libiec61883-0/testing 1.2.0-5 amd64 [upgradable from: 1.2.0-4]
libimage-exiftool-perl/testing 12.42+dfsg-1 all [upgradable from: 12.41+dfsg-1]
libip4tc2/testing 1.8.8-1 amd64 [upgradable from: 1.8.7-1]
libitm1/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libjack-jackd2-0/testing 1.9.21~dfsg-1 amd64 [upgradable from: 1.9.20~dfsg-1+b1]
libjson-c5/testing 0.16-1 amd64 [upgradable from: 0.15-2]
libk5crypto3/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libkadm5clnt-mit12/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libkadm5srv-mit12/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libkdb5-10/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libkeyutils1/testing 1.6.3-1 amd64 [upgradable from: 1.6.1-3]
libkmod2/testing 29-1+b1 amd64 [upgradable from: 29-1]
libkrb5-3/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libkrb5-dev/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
libkrb5support0/testing 1.19.2-2+b2 amd64 [upgradable from: 1.19.2-2+b1]
liblapack3/testing 3.10.1-2 amd64 [upgradable from: 3.10.0-2]
libldap-2.5-0/testing 2.5.12+dfsg-2 amd64 [upgradable from: 2.5.11+dfsg-1]
liblilv-0-0/testing 0.24.14-1 amd64 [upgradable from: 0.24.12-2]
libllvm13/testing 1:13.0.1-6 amd64 [upgradable from: 1:13.0.1-3+b2]
liblsan0/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libltdl7/testing 2.4.7-4 amd64 [upgradable from: 2.4.7-3]
liblua5.4-0/testing 5.4.4-2 amd64 [upgradable from: 5.4.4-1]
liblzma5/testing 5.2.5-2.1 amd64 [upgradable from: 5.2.5-2]
libmariadb3/testing 1:10.6.8-1 amd64 [upgradable from: 1:10.6.7-3]
libmount1/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
libncurses6/testing 6.3+20220423-2 amd64 [upgradable from: 6.3-2]
libncursesw6/testing 6.3+20220423-2 amd64 [upgradable from: 6.3-2]
libneon27/testing 0.32.2-1+b1 amd64 [upgradable from: 0.32.2-1]
libnghttp2-14/testing 1.47.0-1+b1 amd64 [upgradable from: 1.43.0-1]
libnspr4/testing 2:4.34-1 amd64 [upgradable from: 2:4.33-1]
libnss3/testing 2:3.79-1 amd64 [upgradable from: 2:3.77-1]
libogg0/testing 1.3.5-1 amd64 [upgradable from: 1.3.4-0.1]
libonig5/testing 6.9.8-1 amd64 [upgradable from: 6.9.7.1-2]
libopenjp2-7/testing 2.5.0-1 amd64 [upgradable from: 2.4.0-6]
libopenmpt0/testing 0.6.4-1 amd64 [upgradable from: 0.6.2-1]
libopus0/testing 1.3.1-2 amd64 [upgradable from: 1.3.1-0.1]
libp11-kit0/testing 0.24.1-1 amd64 [upgradable from: 0.24.0-6]
libpam-modules-bin/testing 1.4.0-13 amd64 [upgradable from: 1.4.0-11]
libpam-modules/testing 1.4.0-13 amd64 [upgradable from: 1.4.0-11]
libpam-runtime/testing 1.4.0-13 all [upgradable from: 1.4.0-11]
libpam-systemd/testing 251.2-6 amd64 [upgradable from: 250.4-1]
libpam0g/testing 1.4.0-13 amd64 [upgradable from: 1.4.0-11]
libpango-1.0-0/testing 1.50.7+ds-1 amd64 [upgradable from: 1.50.6+ds-2]
libpangocairo-1.0-0/testing 1.50.7+ds-1 amd64 [upgradable from: 1.50.6+ds-2]
libpangoft2-1.0-0/testing 1.50.7+ds-1 amd64 [upgradable from: 1.50.6+ds-2]
libpcre2-8-0/testing 10.40-1 amd64 [upgradable from: 10.39-3]
libpng-dev/testing 1.6.37-5 amd64 [upgradable from: 1.6.37-3]
libpng16-16/testing 1.6.37-5 amd64 [upgradable from: 1.6.37-3]
libpostproc55/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libpulse0/testing 15.0+dfsg1-4+b1 amd64 [upgradable from: 15.0+dfsg1-4]
libpython3-stdlib/testing 3.10.4-1+b1 amd64 [upgradable from: 3.9.8-1]
libpython3.9-minimal/testing 3.9.13-1 amd64 [upgradable from: 3.9.12-1]
libpython3.9-stdlib/testing 3.9.13-1 amd64 [upgradable from: 3.9.12-1]
libquadmath0/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
librabbitmq4/testing 0.11.0-1+b1 amd64 [upgradable from: 0.11.0-1]
libre2-9/testing 20220601+dfsg-1 amd64 [upgradable from: 20220401+dfsg-1]
libreadline8/testing 8.1.2-1.2 amd64 [upgradable from: 8.1.2-1]
librsvg2-2/testing 2.54.4+dfsg-1 amd64 [upgradable from: 2.52.5+dfsg-3+b1]
librsvg2-bin/testing 2.54.4+dfsg-1 amd64 [upgradable from: 2.52.5+dfsg-3+b1]
librubberband2/testing 2.0.2+dfsg0-2 amd64 [upgradable from: 2.0.0-2]
libsasl2-2/testing 2.1.28+dfsg-6 amd64 [upgradable from: 2.1.28+dfsg-2+b1]
libsasl2-modules-db/testing 2.1.28+dfsg-6 amd64 [upgradable from: 2.1.28+dfsg-2+b1]
libsdl2-2.0-0/testing 2.0.22+dfsg-6 amd64 [upgradable from: 2.0.20+dfsg-2]
libseccomp2/testing 2.5.4-1 amd64 [upgradable from: 2.5.3-2]
libselinux1/testing 3.4-1 amd64 [upgradable from: 3.3-1+b2]
libsemanage-common/testing 3.4-1 all [upgradable from: 3.3-1]
libsemanage2/testing 3.4-1 amd64 [upgradable from: 3.3-1+b2]
libsepol2/testing 3.4-2 amd64 [upgradable from: 3.3-1]
libserd-0-0/testing 0.30.12-1 amd64 [upgradable from: 0.30.10-2]
libsigc++-2.0-0v5/testing 2.10.8-1 amd64 [upgradable from: 2.10.4-2]
libslang2/testing 2.3.2-5+b1 amd64 [upgradable from: 2.3.2-5]
libsmartcols1/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
libsnappy1v5/testing 1.1.9-2 amd64 [upgradable from: 1.1.8-1]
libsord-0-0/testing 0.16.10-1 amd64 [upgradable from: 0.16.8-2]
libspeex1/testing 1.2.0-1 amd64 [upgradable from: 1.2~rc1.2-3]
libsqlite3-0/testing 3.38.5-1 amd64 [upgradable from: 3.38.2-1]
libsratom-0-0/testing 0.6.10-1 amd64 [upgradable from: 0.6.8-1]
libsrt1.4-gnutls/testing 1.4.4-4+b1 amd64 [upgradable from: 1.4.4-4]
libssh-gcrypt-4/testing 0.9.6-2+b1 amd64 [upgradable from: 0.9.6-2]
libssh2-1/testing 1.10.0-3+b1 amd64 [upgradable from: 1.10.0-3]
libssl-dev/testing 3.0.3-8 amd64 [upgradable from: 1.1.1n-1]
libssl1.1/testing 1.1.1o-1 amd64 [upgradable from: 1.1.1n-1]
libstdc++-11-dev/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
libstdc++6/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libswresample3/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libswscale5/testing 7:4.4.2-1+b3 amd64 [upgradable from: 7:4.4.1-3+b2]
libsystemd0/testing 251.2-6 amd64 [upgradable from: 250.4-1]
libtdb1/testing 1.4.6-3 amd64 [upgradable from: 1.4.6-1]
libtheora0/testing 1.1.1+dfsg.1-16 amd64 [upgradable from: 1.1.1+dfsg.1-15]
libtiff5/testing 4.4.0-2 amd64 [upgradable from: 4.3.0-6]
libtinfo6/testing 6.3+20220423-2 amd64 [upgradable from: 6.3-2]
libtsan0/testing 11.3.0-3 amd64 [upgradable from: 11.2.0-19]
libubsan1/testing 12.1.0-2 amd64 [upgradable from: 12-20220319-1]
libudev1/testing 251.2-6 amd64 [upgradable from: 250.4-1]
libuuid1/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
libvulkan1/testing 1.3.216.0-1 amd64 [upgradable from: 1.3.204.1-2]
libx264-dev/testing 2:0.164.3095+gitbaee400-2 amd64 [upgradable from: 2:0.163.3060+git5db6aa6-2]
libxcursor1/testing 1:1.2.1-1 amd64 [upgradable from: 1:1.2.0-2]
libxml2-dev/testing 2.9.14+dfsg-1 amd64 [upgradable from: 2.9.13+dfsg-1]
libxml2/testing 2.9.14+dfsg-1 amd64 [upgradable from: 2.9.13+dfsg-1]
libxrandr2/testing 2:1.5.2-2+b1 amd64 [upgradable from: 2:1.5.2-1]
libxrender-dev/testing 1:0.9.10-1.1 amd64 [upgradable from: 1:0.9.10-1]
libxrender1/testing 1:0.9.10-1.1 amd64 [upgradable from: 1:0.9.10-1]
libxtst6/testing 2:1.2.3-1.1 amd64 [upgradable from: 2:1.2.3-1]
libxv1/testing 2:1.0.11-1.1 amd64 [upgradable from: 2:1.0.11-1]
libzimg2/testing 3.0.4+ds1-1 amd64 [upgradable from: 3.0.3+ds1-1+b1]
linux-libc-dev/testing 5.18.5-1 amd64 [upgradable from: 5.16.18-1]
lsb-base/testing 11.2 all [upgradable from: 11.1.0]
lsb-release/testing 11.2 all [upgradable from: 11.1.0]
lsof/testing 4.95.0-1 amd64 [upgradable from: 4.93.2+dfsg-1.1]
mariadb-client-10.6/testing 1:10.6.8-1 amd64 [upgradable from: 1:10.6.7-3]
mariadb-client-core-10.6/testing 1:10.6.8-1 amd64 [upgradable from: 1:10.6.7-3]
mariadb-client/testing 1:10.6.8-1 all [upgradable from: 1:10.6.7-3]
mariadb-common/testing 1:10.6.8-1 all [upgradable from: 1:10.6.7-3]
mawk/testing 1.3.4.20200120-3.1 amd64 [upgradable from: 1.3.4.20200120-3+b1]
media-types/testing 8.0.0 all [upgradable from: 7.1.0]
mount/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
nano/testing 6.3-1 amd64 [upgradable from: 6.2-1]
ncurses-base/testing 6.3+20220423-2 all [upgradable from: 6.3-2]
ncurses-bin/testing 6.3+20220423-2 amd64 [upgradable from: 6.3-2]
nodejs/unknown 16.15.1-deb-1nodesource1 amd64 [upgradable from: 16.14.2-deb-1nodesource1]
openssl/testing 3.0.3-8 amd64 [upgradable from: 1.1.1n-1]
python3-gi/testing 3.42.1-1 amd64 [upgradable from: 3.42.0-3]
python3-minimal/testing 3.10.4-1+b1 amd64 [upgradable from: 3.9.8-1]
python3-pycurl/testing 7.44.1-5 amd64 [upgradable from: 7.44.1-4]
python3.9-minimal/testing 3.9.13-1 amd64 [upgradable from: 3.9.12-1]
python3.9/testing 3.9.13-1 amd64 [upgradable from: 3.9.12-1]
python3/testing 3.10.4-1+b1 amd64 [upgradable from: 3.9.8-1]
readline-common/testing 8.1.2-1.2 all [upgradable from: 8.1.2-1]
rsync/testing 3.2.4-1+b1 amd64 [upgradable from: 3.2.3-8]
shared-mime-info/testing 2.2-1 amd64 [upgradable from: 2.1-2]
sqlite3/testing 3.38.5-1 amd64 [upgradable from: 3.38.2-1]
systemd-sysv/testing 251.2-6 amd64 [upgradable from: 250.4-1]
systemd/testing 251.2-6 amd64 [upgradable from: 250.4-1]
sysvinit-utils/testing 3.03-1 amd64 [upgradable from: 3.01-1]
util-linux/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
uuid-dev/testing 2.38-4 amd64 [upgradable from: 2.37.3-1+b1]
wget/testing 1.21.3-1+b2 amd64 [upgradable from: 1.21.3-1+b1]
x11proto-dev/testing 2022.1-1 all [upgradable from: 2021.5-1]
xz-utils/testing 5.2.5-2.1 amd64 [upgradable from: 5.2.5-2]
```
</details>

At first i thought it's probably chrome (updated from `100.0.4896.88` to `103.0.5060.53`). But if it were chrome, i'd probably see the same change (having the `at` in the string) in the demo, which i don't.
So it's probably one of the other 250 packages.

All this shows a couple of things:
1. The change was not caused by a change in the photoprisms codebase, but rather by an updated dependency within the docker image
2. The environment outside the browser seems to have an impact on that specific date-time-string

I think at the end of the day it doesn't really matter wether the string says `Sunday, July 8, 2012, 2:45 PM` or `Sunday, July 8, 2012 at 2:45 PM`. It wouldn't surprise me if it is already showing the latter one for some users, depending on their environment.

If we don't insist on having that string be equal in all environments, its probably not useful to further investigate.

lastzero commented:
Thanks, that's awesome to know.... Maybe just compare the prefix in this case.
